Automattic/jetpack
# Composer packages: introducing Jetpack Affiliate

Starting with the cool Composer trend. This PR will create a package for the class that reads and Affiliate code. It's later replaced where it was previously used, to add the affiliate code to the connection URL.

#### Changes proposed in this Pull Request:

* Move the `Jetpack_Affiliate` class to its own Composer package.
* in files where this class was used, import the namespaced class
* small tweaks in Jetpack_Affiliate class so it's PHPCS compliant

#### Testing instructions:






* checkout the branch
* do `composer require automattic/jetpack-affiliate`
* test manually going to WP Admin > Dashboard in a _disconnected_ site 
  1. do `wp option add jetpack_affiliate_code abc123` (or `yarn docker:wp option add jetpack_affiliate_code abc123`)
  2. verify that the URL of the Set up Jetpack button has the `aff` parameter with the right value
* run unit tests from `affiliate` suite
  1. run `phpunit --testsuite=affiliate` (or `yarn docker:phpunit --testsuite=affiliate`)

#### Proposed changelog entry
* Move Jetpack Affiliate code to a Composer package.
